src/main/java/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package suppliers;
+
+import core.basesyntax.model.Circle;
+import core.basesyntax.model.Figure;
+import core.basesyntax.model.IsoscelesTrapezoid;
+import core.basesyntax.model.Rectangle;
+import core.basesyntax.model.RightTriangle;
+import core.basesyntax.model.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(4);
+        switch (index) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), getRandomInt());
+            case 1:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        getRandomDouble(), getRandomDouble(), getRandomDouble());
+            case 2:
+                return new Rectangle(colorSupplier.getRandomColor(), getRandomDouble(),
+                        getRandomDouble());
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(), getRandomInt(),
+                        getRandomInt());
+            case 4:
+                return new Square(colorSupplier.getRandomColor(), getRandomDouble());
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    private double getRandomDouble() {
+        return new Random().nextDouble();
+    }
+
+    private int getRandomInt() {
+        return new Random().nextInt(100);
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("white", 10);
+    }
```

Comment:
100, 10, "white" should be constants
public methods should be over private

Reply:
Done